src/System.Management.Automation/engine/CommandCompletion/CompletionHelpers.cs

```diff
@@ -14,43 +14,142 @@ internal static class CompletionHelpers
     {
         private static readonly SearchValues<char> s_defaultCharsToCheck = SearchValues.Create("$`");
 
+        private const string SingleQuote = "'";
+        private const string DoubleQuote = "\"";
+
         /// <summary>
         /// Get matching completions from word to complete.
         /// This makes it easier to handle different variations of completions with consideration of quotes.
         /// </summary>
         /// <param name="wordToComplete">The word to complete.</param>
         /// <param name="possibleCompletionValues">The possible completion values to iterate.</param>
         /// <param name="toolTipMapping">The optional tool tip mapping delegate.</param>
+        /// <param name="listItemTextMapping">The optional list item text mapping delegate.</param>
         /// <param name="resultType">The optional completion result type. Default is Text.</param>
-        /// <returns></returns>
+        /// <param name="matchStrategy">The optional match strategy delegate.</param>
+        /// <returns>List of matching completion results.</returns>
         internal static IEnumerable<CompletionResult> GetMatchingResults(
             string wordToComplete,
             IEnumerable<string> possibleCompletionValues,
             Func<string, string> toolTipMapping = null,
-            CompletionResultType resultType = CompletionResultType.Text)
+            Func<string, string> listItemTextMapping = null,
+            CompletionResultType resultType = CompletionResultType.Text,
+            MatchStrategy matchStrategy = null)
         {
+            matchStrategy ??= DefaultMatch;
+
             string quote = HandleDoubleAndSingleQuote(ref wordToComplete);
-            var pattern = WildcardPattern.Get(wordToComplete + "*", WildcardOptions.IgnoreCase);
+            if (quote == DoubleQuote)
```

Comment:
If there is not a quote (1) input string is interpreted as expandable (double quoted) by Tokenizer, (2) HandleDoubleAndSingleQuote return empty string in the case. So with the condition we don't catch all we need.

HandleDoubleAndSingleQuote return single quote only if it is explicitly used. Yes?
So I think the check should be:
```suggestion
            if (quote != SingleQuote)
```

Reply:
@iSazonov Yep makes sense actually, commited the change.